### ISSUE TITLE
Pastiche Master: Exemplar-Based High-Resolution Portrait Style Transfer

### ISSUE THREAD

**Issue by hypknot74** [hypknot74/Paper-summaries#36]
全体的に理解できていない箇所がある

### リンク
arxiv: [https://arxiv.org/abs/2203.13248](https://arxiv.org/abs/2203.13248)
project page: [https://www.mmlab-ntu.com/project/dualstylegan/](https://www.mmlab-ntu.com/project/dualstylegan/)
github: [https://github.com/williamyang1991/DualStyleGAN](https://github.com/williamyang1991/DualStyleGAN)

### 著者/所属機関
Shuai Yang, Liming Jiang, Ziwei Liu, Chen Change Loy
S-Lab, Nanyang Technological University

### 投稿日付
CVPR2022

## 概要
target domainのcolorとstructrueをdisentangleして，FFHQ pre-trained StyleGANに階層的に情報を加えることで画像変換を可能にするDualStyleGANを提案．

## 手法
stylize: 様式化 (絵画化)
destylize: 非様式化 (実写化)
g: FFHQ pre-trained StyleGAN
g’: fine-tuned StyleGAN
S: 入力絵画画像
$z^+_e$: エンコーダでZ+空間に埋め込んだ潜在変数
$\hat{z}^+_e$: よりdestylizeするために$z^+_e$をg’を使用して最適化した潜在変数
$z^+_i$: $E(g(\hat{z}^+_e))$によってさらにdestylizeした潜在変数

### Facial Destylization (絵画の実写化)
![image](https://user-images.githubusercontent.com/57225520/202848339-b0c0f541-480b-4409-a606-91ee74265364.png)
**Stage 1: 潜在変数の初期化**
FFHQで事前学習したZ+空間にinversionするようにpSpを修正したエンコーダEで潜在変数を求める．FFHQで事前学習したStyleGANを使用するが，色と構造は入力絵画画像Sを良く捉えられる．
図の(b)がstage1で得られる画像．

**stage 2: 潜在変数の最適化**
より実写に近い再構成画像を得るために，stage 1で得られたz+をfine-tuningしたg’を用いて以下の問題で最適化する．
![image](https://user-images.githubusercontent.com/57225520/202848349-6e716fcf-7302-4e3d-932a-f15bdc4de6e1.png)
$||\sigma(z^+)||_1$ は18個の512次元の潜在変数の標準誤差．これを加えることで入力絵画画像に過剰適合しないようにする．
図の(c)がstage 2で得られる画像．この最適化を行わないと(g)のように過剰適合する．また，Z+ではなく平均潜在変数で初期化すると(j)のように不十分な再構成になる．

疑問点
なぜfine-tuningしたg’を使用して最適化することで，再構成画像は実写に近づくのか．むしろ入力絵画画像Sに近づくのではないか．
g’を使用して最適化することで，より自然な空間(分布内)に最適化され，gに入力した画像が分布内の実写に近づくということ？

**stage 3: 画像の埋め込み**
より実写顔画像に近づけるために， $\hat{z}^+_e$ をgに入力して生成した画像をさらにエンコーダEに入れて埋め込むことで $z^+_i$ を得る．
$z^+_i = E(g(\hat{z}^+_e))$

### DualStyleGAN
![image](https://user-images.githubusercontent.com/57225520/202848374-011c3176-2c2b-44e6-ae80-89e106204a75.png)
**Color control**
$z^+_e$をマッピングネットワークfを通じてTcに入力して，ドメイン固有の色に変換する．これをfine-resolution layersのAdaINにおいて重みwで組み合わせる．

**Structure control**
Tsを通じてドメイン固有の構造に変換し，それをcoarse-resolution layersのmodulative residual blocks (ModRes)に入力して重みwで組み合わせる．ModResではResBlockを使用しており，AdaINでstyleを与える．

fine-tuningにおいてStyleGANのサブモジュール(図のGeneratorの緑の畳み込み層)が最も変化するため，そこは固定して，畳み込み層における変化のみをModResで学習させる．そうすることで，fine-tuningしてもなるべく元のネットワークの特徴量と対応付けることができ，実写と絵画で共有潜在空間を構築できる．

StyleGANにおける一般的なmodulationにchannel-wise, spatial-wise, element-wiseがあり，それぞれ，AdaIN，Diagonal Attention(DAT)，ResBlockに対応する．これらを使用して実験したところ，ResBlockが最もToonify(完全にfine-tuningした場合)に近かった．よって，ModResにResBlockとAdaINを採用した．
![image](https://user-images.githubusercontent.com/57225520/202848390-dcd7df06-fe99-4fe1-86d0-102de818538e.png)

### Progressive Fine-Tuning
この図の(a)が何を示すのかよくわからない．
![image](https://user-images.githubusercontent.com/57225520/202848411-3454904c-552f-46f1-a0e5-169b98ce3eb7.png)
**Stage 1: Color transfer on source domain**
特定のパラメタの初期化のみを行う．ModResの値は0に設定．Tcの全結合層の値も単位行列に設定し，入力潜在変数に変化を加えない．つまり，StyleGANのstyle mixingと同じことをしているだけで，構造を保ちつつ色だけ絵画を反映させる．

**Stage 2: Structure transfer on source domain**
化粧のような中間レベルのスタイルを変換できるように学習する．z1とz2はランダムな潜在変数なので，特に絵画の情報を学習しているわけではない．(すなわち，DualStyleGANでStyle Mixingと同じことをできるように学習させているのと同じ)
![image](https://user-images.githubusercontent.com/57225520/202848424-6173cdc5-4765-42d6-96c3-6155ef107219.png)
G: DualStyleGANのGenerator
g: FFHQ pre-trained StyleGAN
z1, z2: ランダムな潜在変数
$\tilde{z}_2$ : $\{z_2 , E(g(z_2))\}$ からサンプリングした潜在変数．(なにが目的？)
$z^+_l$ : z1のlベクトルとz2の(18-l)ベクトルを結合したベクトル．これはStyle Mixingと同じ．lはfine-tuning時に7から5まで減らしていく．つまり，段々と絵画のスタイルに寄せていく．

**stage 3: Style transfer on target domain**
ようやくDualStyleGANを絵画(target domain)でfine-tuningする．
![image](https://user-images.githubusercontent.com/57225520/202848439-2b1e3a9d-af71-47b1-8eee-a07c8360f779.png)

$L_{perc}$  : 絵画画像Sから $z^+_i$ と $z^+_e$ を得て， $L_{perc}(G(z^+_i , z^+_e , 1), S)$ を求めたもの
![image](https://user-images.githubusercontent.com/57225520/202848493-cbba54e1-6581-4322-a072-bb9417961d7d.png)
$L_{CX}$ : contextual loss (参考: The contextual loss for image transformation with non-aligned data.)
$L_{FM}$ : feature matching loss (参考: Arbitrary style transfer in real-time with adaptive instance normalization.)
![image](https://user-images.githubusercontent.com/57225520/202848516-b9329596-28d4-41ef-82f0-68b3678cc3bf.png)
### 潜在変数の最適化
target domainにも多様なスタイルが含まれており，お手本とする画像のスタイルにより近づけるために，perceptual lossとcontextual lossを最小化するように潜在変数を最適化する．
![image](https://user-images.githubusercontent.com/57225520/202848530-1f31f91f-e9d2-4f3f-8270-0731f3a5d4ea.png)

### 潜在変数のサンプリング
ガウスノイズを最適化されたextrinsic style codeの分布に揃えるためにネットワークNを学習する．structure codeとcolor codeは独立してNからサンプリングされ，結合してextrinsic style codeを構成する．(つまり，図ではcolor codeをTsに入れてstructre codeに変換しているように見えるが，実際はNからサンプリングされてマッピングネットワークを通したstructure codeのみがTsに入力されている？)

## 結果
![image](https://user-images.githubusercontent.com/57225520/202848557-3b6bd12d-8035-4bc6-8f31-f99554eb97dc.png)
![image](https://user-images.githubusercontent.com/57225520/202848580-c9eeb84d-2049-49ff-a321-825cb4ba23c1.png)
![image](https://user-images.githubusercontent.com/57225520/202848567-03c144b5-39fb-493a-93ad-dc7e3a6914e4.png)
(a)はFacial Destylizationの有無について．(Facial Destylizationが結局どのように使用されているのかわからない)
![image](https://user-images.githubusercontent.com/57225520/202848588-dec60849-1e96-4424-b8a5-e356d2b9ffc0.png)

## コメント
Facial Destylizationは白黒漫画でどれだけできるのだろう．
損失関数のハイパラ調整が難しそう．
